shared/src/metabase/mbql/util.cljc

```diff
@@ -377,15 +377,12 @@
   "Dispatch function perfect for use with multimethods that dispatch off elements of an MBQL query. If `x` is an MBQL
   clause, dispatches off the clause name; otherwise dispatches off `x`'s class."
   ([x]
-   #?(:clj
-      (if (mbql-clause? x)
-        (first x)
-        (or (metabase.models.dispatch/model x)
-            (type x)))
-      :cljs
-      (if (mbql-clause? x)
-        (first x)
-        (type x))))
+   (if (mbql-clause? x)
+     (first x)
+     (if (keyword? x)
+       x
+       (or #?(:clj (metabase.models.dispatch/model x))
```

Comment:
This allows the opportunity to dispatch on special keywords like the new `::sql.qp/substring-name`.